odxtools/minmaxlengthtype.py

```diff
@@ -53,94 +53,125 @@ def __termination_character(self):
 
     def convert_internal_to_bytes(self, internal_value, encode_state: EncodeState,
                                   bit_position: int) -> bytes:
-        byte_length = self._minimal_byte_length_of(internal_value)
-
-        # The coded value must have at least length min_length
-        if byte_length < self.min_length:
-            raise EncodeError(f"The internal value {internal_value} is only {byte_length} bytes"
-                              f" long but the min length is {self.min_length}")
-        # The coded value must not have a length greater than max_length
-        if self.max_length and byte_length > self.max_length:
-            raise EncodeError(f"The internal value {internal_value} requires {byte_length}"
-                              f" bytes, but the max length is {self.max_length}")
-
-        value_byte = self._to_bytes(
-            internal_value,
-            bit_position=0,
-            bit_length=8 * byte_length,
-            base_data_type=self.base_data_type,
-            is_highlow_byte_order=self.is_highlow_byte_order,
-        )
-
-        if encode_state.is_end_of_pdu or byte_length == self.max_length:
-            # All termination types may be ended by the PDU
-            return value_byte
+        bit_length = 8 * self.max_length if self.max_length is not None else 8 * len(internal_value)
+        bit_length = min(8 * len(internal_value), bit_length)
+        value_bytes = bytearray(
+            self._to_bytes(
```

Comment:
internal value can be a variable size int, not just bytes and str

Reply:
the spec only allows standard length types here, and even these tend to come with severe restrictions (e.g. for A_FLOAT32 min_length == max_length == 32.) AFAICS, the only data type which has variable size are strings, cf. section 7.3.6.2 of the spec (p. 94 ff)

(Honestly, I have severe doubts about the usefulness of MinMaxLengthType in practice...)

Reply:
Usually there is no termination sequence for variable size int, only valid value I can think of would be end-of-pdu.

Reply:
yeah, I suppose that being unable to specify integers that have their most-significant byte 0 is kind of a show stopper...

that said, I believe `._to_bytes()` works for arbitrary data types? (at least modulo bugs)